Weather/app/src/main/java/com/example/weather/City.java

```diff
@@ -0,0 +1,63 @@
+package com.example.weather;
+
+import android.content.Intent;
+import android.os.Bundle;
+import android.support.v7.app.AppCompatActivity;
+import android.view.View;
+import android.widget.Button;
+import android.widget.Toast;
+
+public class City extends AppCompatActivity implements View.OnClickListener {
+
+    Button button;
+    Button button2;
+    Button button3;
+    Button button4;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_city);
+
+        button = (Button) findViewById(R.id.button);
+        button.setOnClickListener(this);
+
+        button2 = (Button) findViewById(R.id.button2);
+        button2.setOnClickListener(this);
+
+        button3 = (Button) findViewById(R.id.button3);
+        button3.setOnClickListener(this);
+
+        button4 = (Button) findViewById(R.id.button4);
+        button4.setOnClickListener(this);
+    }
+
+    @Override
+    public void onClick(View v) {
+        switch (v.getId()) {
```

Comment:
```suggestion
 string index = "";
switch(v.getId()){
case R.id.button:
 index = "1";
 break;
case R.id.button2: .....
....
}
Toast.makeText(this, String.format("Нажата кнопка %s", index) ....;
Intent intent...
```

То есть вынести повторяющийся код за switch